src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    for (const key in action) {
```

Comment:
That's a useless loop, that makes the complexity of your solution not that great.